geps/gep-1619.md

```diff
@@ -60,26 +71,162 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
+
+Here an example implementation of a gateway initiating a session through cookies:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B-->>-G: Response
+    G->>G: Add set-cookie header
+    G-->>-C: Response<br>[set-cookie]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header.<br>May include other set-cookie<br>headers from backend.
+    C->>C: Create Cookie(s)<br>from set-cookie header(s)
+    Note right of C: [cookie] indicates a request<br> with one or more cookies
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup of<br>server using cookie value
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+#### Backend Initiated Session Example
+
+**Important**: While we took it into consideration, this GEP does not support configuring backend-initiated sessions.
+This could potentially affect frameworks that initiate sessions in the backend. Implementing this feature is complicated
+and requires careful design, making it suitable for exploration in a separate GEP.
+
+Continuing with the cookie example, when dealing with backend-initiated sessions, the process becomes somewhat more
+complex. For cookie-based session persistence, the gateway needs store a value within a cookie containing a server
+identifier. This identifier can be then used as a reference to maintain a persistent session to a specific backend.
+There are several approaches a gateway could use in this situation to achieve session persistence:
+
+1. Modify the existing cookie's value
+2. Insert an additional cookie
+3. Prefix the existing cookie
+
+Alternatively, if the backend is not configured for session persistence, the gateway should refrain from modifying or
+inserting a cookie. In this situation, the gateway should remain passive and simply forward the `set-cookie` header as
+it is.
+
+Refer to the [Cookie Rewriting](#cookie-rewriting) section of the API for implementation guidance.
+
+Here's an example implementation of a backend initiating a session and the gateway modifies the cookie:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B->>B: Add set-cookie<br>header
+    B-->>-G: Response<br>[set-cookie]
+    G->>G: Modify set-cookie<br>header per configuration
+    G-->>-C: Response<br>[set-cookie*]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header<br>[set-cookie*] indicates a response<br>with a MODIFIED set-cookie header
+    C->>C: Create Cookie<br>from set-cookie header
+    Note right of C: [cookie] indicates a request<br>or response with a cookie
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup<br>of server using cookie value
+    G->>+B: Request<br>[cookie]
```

Comment:
> It is unclear if the gateway needs to "rewrite" the cookie back to something that the backend understands, otherwise the backend should not use the contents of the cookie for any internal use to identify the session.
IMHO allowing the modification of a cookie creates a high coupling between the backend and gateway implementations, and I'm not sure this is advisable in the general case.

These are good points. As a disclaimer, the current state of this GEP is that it's not attempting to prescribe how implementations should handle backend initiated sessions as stated in the "Session Initiation Guidelines" section: `We leave the decision of handling existing sessions with each specific implementation.`. This section wasn't intending to prescribe anything, but to explore potential solutions, though I understand exploring examples can be suggestive in itself.

Given that, I reorganized the list, added a statement that there are nuances to inserting/rewriting, and mentioned that inserting a cookie is a safe option, while still allowing implementation to use their discretion. Let me know if this addresses your concerns.